File: problem-083/solution.py
class TreeNode:
    def __init__(self, val: int, left: 'TreeNode' = None, right: 'TreeNode' = None) -> None:
        self.val = val
        self.left = left
        self.right = right

def invert_tree(root: TreeNode) -> TreeNode:
    if not root:
        return root
    
    root.left, root.right = invert_tree(root.right), invert_tree(root.left)
    return root

def is_same_tree(a: TreeNode, b: TreeNode) -> bool:
    if not a and not b:
        return True
    
    if not a or not b:
        return False

    if a.val != b.val:
        return False

    return is_same_tree(a.left, b.left) and is_same_tree(a.right, b.right)    

test_node = TreeNode(5, TreeNode(3, TreeNode(1), TreeNode(4)), TreeNode(10, None, TreeNode(18)))
expected = TreeNode(5, TreeNode(10, TreeNode(18)), TreeNode(3, TreeNode(4), TreeNode(1)))
inverted_tree = invert_tree(test_node)
assert is_same_tree(expected, inverted_tree)

File: problem-083/solution.go
package problem083

type treeNode struct {
	val   int
	left  *treeNode
	right *treeNode
}

// invertTree inverts a binary tree in-place.
func invertTree(root *treeNode) {
	if root == nil {
		return
	}

	invertTree(root.left)
	invertTree(root.right)
	root.left, root.right = root.right, root.left
}
